libs/test/SalesforceSDKTest/src/com/salesforce/androidsdk/auth/idp/IDPSPManagerTestCase.kt

```diff
@@ -0,0 +1,50 @@
+package com.salesforce.androidsdk.auth.idp
+
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import android.util.Log
+import androidx.test.platform.app.InstrumentationRegistry
+import com.salesforce.androidsdk.accounts.UserAccount
+import com.salesforce.androidsdk.util.LogUtil
+import org.junit.Assert
+import java.util.concurrent.ArrayBlockingQueue
+import java.util.concurrent.BlockingQueue
+import java.util.concurrent.TimeUnit
+
+internal open class IDPSPManagerTestCase {
+    companion object {
+        const val MAX_EVENTS = 16
+        const val TIMEOUT: Long = 3
+    }
+
+    lateinit var context: Context
+    lateinit var recordedEvents: BlockingQueue<String>
+
+    open fun setup() {
+        context = InstrumentationRegistry.getInstrumentation().targetContext
+        recordedEvents = ArrayBlockingQueue(MAX_EVENTS)
+    }
+
+    fun sendBroadcast(context:Context, intent: Intent) {
```

Comment:
The IDPManager and SPManager uses that function for sending broadcast. The intent is not actually sent out but it is recorded in a blocking queue that the test waits on.